### ISSUE TITLE
(Feature request) Export settings files (k9s) compatible with Thunderbird and Evolution

### ISSUE THREAD

**Issue by mkrupczak3** [k9mail/k-9#4777]
### Expected behavior
K9-mail account settings backups and imports should be compatible with their counterparts on popular Evolution Mail and Thunderbird email clients

### Actual behavior
K9-mail account settings export files (.k9s) are not compatible with the evolution email client, nor vice versa I believe.

### Steps to reproduce
1. Export account settings from K9-mail by tapping the kebab menu from the main screen -> export settings and accounts -> save
2. Transfer the saved k9s file to a computer with Evolution Mail and/or Thunderbird installed
3. Evolution Mail and/or Thunderbird do not accept this settings export file format (I believe).

### Environment
K-9 Mail version:
5.600

Android version:
9

Account type (IMAP, POP3, WebDAV/Exchange):
Yes

Please take some time to [retrieve logs](https://github.com/k9mail/k-9/wiki/LoggingErrors) and attach them here:
Not applicable

**Comment by mkrupczak3** [comment]
Issue counterpart in Evolution Mail GitLab:
https://gitlab.gnome.org/GNOME/evolution/-/issues/940

**Comment by cketti** [comment]
Can you please describe the problem you're trying to solve.

**Comment by mkrupczak3** [comment]
Hi @cketti,

This is kind of a small quality of life improvement request for users, I.e. to be able to import/export account settings in between popular open source email clients to prevent users from needing to manually re-enter account settings for every account

**Comment by mkrupczak3** [comment]
For instance, the problem I had was that I had 4-5 email accounts configured on k9 mail, but to transfer them over to Evolution or Thunderbird I would have to retype all the settings manually instead of just being able to export from k9 and import into Evolution/Thunderbird.

I'm not sure how many users would have a similar issue, considering that most people use Gmail or 365 or whatever nowadays, but I thought I might as well file this issue.

**Comment by dilyanpalauzov** [comment]
What does K9 do with WebDAV?

As outlined at https://gitlab.gnome.org/GNOME/evolution/-/issues/940, K9 7.xxx does implement the Thunderbird autoconfiguration mechanism. Once it is ready in K9 (and also now), domain owners can setup the webservers accordingly, users type in the account setup in the MUA (K9, Thunderbird, Evolution) `email@domain` and the MUA magically finds the settings.  There will be no need to import/export settings, except when the server is not ready for the `Autoconfiguration` mechanism.

**Comment by cketti** [comment]
If we're only talking about server settings, I guess we could come up with a common file format to store them. It might not be too much work if we base it on Thunderbird's autoconfiguration [config file format](https://wiki.mozilla.org/Thunderbird:Autoconfiguration:ConfigFileFormat).

There's the question of what to do with passwords. Including them in plain text feels like a bad idea, especially if we write settings to a simple file and leave it up to the user to get them from one device to the other. They will end up being sent around via unencrypted email, or copied to some cloud storage etc.
Not including passwords is annoying if you have a lot of accounts. Having a large number of accounts is not too common overall, but probably quite common for users of this hypothetical functionality.

Having a file the user has to get from A to B doesn't make for a great user experience. It would be much nicer if email apps did all the work. But for that you'd need some kind of shared storage. I can think of two ways:
1. Have a main email account that you set up in your client and that contains the server settings to all other accounts in a defined location on the mail server. This only works for protocols that allow uploading messages, i.e. not with POP3 accounts.
2. Use a different service to store server settings. I talked with @ryanleesipes about this in the past. He suggested using [Firefox Sync](https://en.wikipedia.org/wiki/Firefox_Sync). It allows storing arbitrary content. And end-to-end encryption is an integral part of the protocol. The server is [open source](https://github.com/mozilla-services/syncserver) and users can host their own.

Of course using a server to sync the data comes with its own set of challenges. Who's allowed to write data when? What to do when updating a file that contains settings the client doesn't understand?

If we're talking about sharing app configuration, things get a whole lot more complicated.

**Comment by mkrupczak3** [comment]
All good ideas. I should check out the Firefox Sync for personal stuff as well I guess. 

Honestly, I think the re-typing password parts isn't too difficult. The real challenge (especially for non-technical users) is all the IMAP/account settings and all that crap. Most normal humans are scared to death of that stuff. 

My thinking is, 
1. Say your IT department or tech wiz idiot nephew sets up your phone with k9mail
2. You should be able to export/import a config file from that and into Thunderbird (chiefly) and Evolution (my case) or Outlook (can one dream?) such that the user doesn't have to think about IMAP settings
3. If a non-technical user has to think about IMAP settings at any point in this process, then it has failed its purpose.

**Comment by mkrupczak3** [comment]
Is there a page anywhere with the file format definition for the k9s file?

https://gitlab.gnome.org/GNOME/evolution/-/issues/940#note_805696

**Comment by mkrupczak3** [comment]
Also, it appears that Thunderbird and Evolution use a common [autoconfig](https://gitlab.gnome.org/GNOME/evolution/-/issues/940#note_805668) WebDav thingy that is possible for server owners to set up that k9mail "almost implemented for the 5.7.xx versions". I should tell the owner of my email server to add this functionality to reduce user friction.

**Comment by cketti** [comment]
> If a non-technical user has to think about IMAP settings at any point in this process, then it has failed its purpose.

I agree. But this proposal is not going to fix that. You'd still need at least one client to be set up by an "expert". The way to do it is to make setting up an account easier, not transferring server settings. See issue #865.

> Is there a page anywhere with the file format definition for the k9s file?

No. But this file also contains much more than server settings. It's not a suitable format for exchanging server settings with other clients.

**Comment by mkrupczak3** [comment]
Ah, okay. 

So, apparently with Evolution, there's something called [Autoconfig](https://wiki.gnome.org/Apps/Evolution/Autoconfig) available for sysadmins to set up on email servers (I believe) where it provides a standard WebDav location where settings are downloaded and applied for the client. 

This sounds like a really nice solution to prevent users from having to dig into the scary internals of email protocols I feel. 

It's also a bit of sprinkles on top of a crap sandwich though (with the whole email config situation). So, I'm not sure how I feel about that. 

Anyways, the issue I submitted for Evolution was labeled as "won't fix" due to the perceived difficulty of keeping consistency with importing/exporting settings files.

**Comment by mkrupczak3** [comment]
I would guess that this issue should either be closed, or some kind of usable interchange format should be dusted off/invented/etc.
